metrics/segmentStats.py

```diff
@@ -0,0 +1,506 @@
+
+        Parameters
+        ----------
+        subGraphSkeleton: Networkx Graph
+            disjoint subgraph of the skeleton's networkx graph
+
+        cycle: list
+            list of nodes in the cycle
+
+        Returns
+        -------
+        subGraphSkeleton : Networkx graph
+            graph changed inplace with visited nodes removed
+
+        countDict: Dictionary
+            class instance variable changed in place with key as source node,
+            value is the number of segments attached to the source node
+
+        lengthDict: Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the length of the segment
+
+        tortuosityDict: Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the tortuosity of the segment (Nan but saved as zero)
+
+        contractionDictdict: Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the contraction of the segment (0) since it is a cycle
+
+        """
+        listOfPerms = list(itertools.permutations(self._branchPoints, 2))
+        for sourceOnTree, item in listOfPerms:
+            if nx.has_path(subGraphSkeleton, sourceOnTree, item) and sourceOnTree != item:
+                simplePaths = list(nx.all_simple_paths(subGraphSkeleton, source=sourceOnTree, target=item))
+                for simplePath in simplePaths:
+                    simplePath
+                    if sum([1 for point in simplePath if point in self._branchPoints]) == 2:
+                        if sourceOnTree not in self._visitedSources:
+                            self.countDict[sourceOnTree] = 1
+                            self._visitedSources.append(sourceOnTree)
+                        else:
+                            self.countDict[sourceOnTree] = self.countDict[sourceOnTree] + 1
+                        curveLength = self._getLengthAndRemoveTracedPath(subGraphSkeleton, simplePath)
+                        curveDisplacement = np.sqrt(np.sum((np.array(sourceOnTree) - np.array(item)) ** 2))
+                        self.lengthDict[self.countDict[sourceOnTree], sourceOnTree, item] = curveLength
+                        self.tortuosityDict[self.countDict[sourceOnTree], sourceOnTree, item] = curveLength / curveDisplacement
+                        self.contractionDict[self.countDict[sourceOnTree], sourceOnTree, item] = curveDisplacement / curveLength
+                        if curveDisplacement != 0.0:
+                            if np.log(curveDisplacement) != 0.0:
+                                self.hausdorffDimensionDict[self.countDict[sourceOnTree], sourceOnTree, item] = np.log(curveLength) / np.log(curveDisplacement)
+
+    def setStats(self):
```

Comment:
@belisarius222 :  You are right about this function calling the other function. The docstring for this function completely missed my mind. Will write that. It would not be possible to split this up to smaller functions though